Undefined symbols when build with CMake

Step to reproduce:

```
rm -rf build
mkdir -p build
cd build
cmake -DCMAKE_PREFIX_PATH=$HOME/src/github.com/wingyplus/jeab/.deps ..
make
```

It's show compilation fail below:

```
Undefined symbols for architecture x86_64:
  "_IMG_InitJPG", referenced from:
      _IMG_Init in IMG.c.o
  "_IMG_InitPNG", referenced from:
      _IMG_Init in IMG.c.o
  "_IMG_InitTIF", referenced from:
      _IMG_Init in IMG.c.o
  "_IMG_Load", referenced from:
      _IMG_LoadTexture in IMG.c.o
     (maybe you meant: _IMG_LoadXXX_RW, _IMG_LoadICO_RW , _IMG_LoadLBM_RW , _IMG_LoadTexture , _IMG_LoadGIFAnimation_RW , _IMG_LoadBMP_RW , _IMG_LoadGIF_RW_Internal , _IMG_LoadPNM_RW , _IMG_LoadAnimationTyped_RW , _IMG_LoadTextureTyped_RW , _IMG_LoadXV_RW , _IMG_LoadTexture_RW , _IMG_LoadWEBP_RW , _IMG_LoadAnimation , _IMG_LoadTyped_RW , _IMG_LoadPCX_RW , _IMG_LoadAnimation_RW , _IMG_LoadSVG_RW , _IMG_LoadCUR_RW , _IMG_LoadXPM_RW , _IMG_LoadXCF_RW , _IMG_Load_RW )
  "_IMG_LoadGIF_RW", referenced from:
      _supported in IMG.c.o
     (maybe you meant: _IMG_LoadGIF_RW_Internal)
  "_IMG_LoadJPG_RW", referenced from:
      _supported in IMG.c.o
  "_IMG_LoadPNG_RW", referenced from:
      _supported in IMG.c.o
  "_IMG_LoadTGA_RW", referenced from:
      _supported in IMG.c.o
  "_IMG_LoadTIF_RW", referenced from:
      _supported in IMG.c.o
  "_IMG_QuitJPG", referenced from:
      _IMG_Quit in IMG.c.o
  "_IMG_QuitPNG", referenced from:
      _IMG_Quit in IMG.c.o
  "_IMG_QuitTIF", referenced from:
      _IMG_Quit in IMG.c.o
  "_IMG_isGIF", referenced from:
      _supported in IMG.c.o
      _supported_anims in IMG.c.o
  "_IMG_isJPG", referenced from:
      _supported in IMG.c.o
  "_IMG_isPNG", referenced from:
      _supported in IMG.c.o
  "_IMG_isTIF", referenced from:
      _supported in IMG.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libSDL2_image.dylib] Error 1
make[1]: *** [CMakeFiles/SDL2_image.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,15 +10,31 @@ option(SUPPORT_PNG "Support loading PNG images" ON)
 option(SUPPORT_WEBP "Support loading WEBP images" OFF)
 option(BUILD_SHOWIMAGE "Build the showimage sample program" OFF)
 option(BUILD_SHARED_LIBS "Build the library as a shared library" ON)
+if (APPLE)
+	option(ENABLE_APPLE_IMAGEIO "Use native Mac OS X frameworks for loading images" ON)
+endif()
+
 
 if (NOT BUILD_SHARED_LIBS)
 	set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()
 
 add_library(SDL2_image)
+
+set(IMAGEIO_SOURCES)
+if (APPLE)
+	if (ENABLE_APPLE_IMAGEIO)
+		target_link_options(SDL2_image PRIVATE -Wl,-framework,ApplicationServices)
+		target_link_libraries(SDL2_image PRIVATE objc)
+		set(IMAGEIO_SOURCES "IMG_ImageIO.m")
+	else()
+		target_compile_definitions(SDL2_image PRIVATE SDL_IMAGE_USE_COMMON_BACKEND)
+	endif()
+endif()
+
 target_sources(SDL2_image PRIVATE IMG.c IMG_png.c IMG_bmp.c IMG_gif.c
 		IMG_jpg.c IMG_lbm.c IMG_pcx.c IMG_pnm.c IMG_svg.c IMG_tga.c
-		IMG_tif.c IMG_webp.c IMG_WIC.c IMG_xcf.c IMG_xpm.c IMG_xv.c IMG_xxx.c)
+		IMG_tif.c IMG_webp.c IMG_WIC.c IMG_xcf.c IMG_xpm.c IMG_xv.c IMG_xxx.c ${IMAGEIO_SOURCES})
 
 target_compile_definitions(SDL2_image PRIVATE
 		-DLOAD_BMP -DLOAD_GIF -DLOAD_LBM -DLOAD_PCX -DLOAD_PNM

